### PR TITLE
feat(mesh catalog): Return list of services for a given service account

### PR DESCRIPTION
**Pull request by ksubrmnn**
Please describe the motivation for this PR and provide enough
information so that others can review it.

Returns a list of services instead of a single service for a given service account 

Please mark with X for applicable areas.

- New Functionality      [X]
- Documentation          [ ]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [X]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests / CI System      [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

No

### CODE REVIEW

File: pkg/catalog/service.go [hunk 1]
```diff
@@ -5,16 +5,26 @@ import (
...
-// GetServiceForServiceAccount returns a service corresponding to a service account
-func (mc *MeshCatalog) GetServiceForServiceAccount(sa service.K8sServiceAccount) (service.MeshService, error) {
+// GetServicesForServiceAccount returns a list of services corresponding to a service account
+func (mc *MeshCatalog) GetServicesForServiceAccount(sa service.K8sServiceAccount) ([]service.MeshService, error) {
+	services := []service.MeshService{}
 	for _, provider := range mc.endpointsProviders {
 		// TODO (#88) : remove this provider check once we have figured out the service account story for azure vms
 		if provider.GetID() != constants.AzureProviderName {
 			log.Trace().Msgf("[%s] Looking for Services for Name=%s", provider.GetID(), sa)
 			service, err := provider.GetServiceForServiceAccount(sa)
```
Comment (ksubrmnn) [inline pkg/catalog/service.go:15]: will refactor provider functions (if needed) when Shashank merges his change

File: pkg/catalog/routes.go [hunk 2]
```diff
@@ -237,28 +237,30 @@ func getTrafficPolicyPerRoute(mc *MeshCatalog, routePolicies map[trafficpolicy.T
 			Namespace: trafficTargets.Spec.Destination.Namespace,
 			Name:      trafficTargets.Spec.Destination.Name,
 		}
-		destService, destErr := mc.GetServiceForServiceAccount(dstNamespacedServiceAcc)
+		destServiceList, destErr := mc.GetServicesForServiceAccount(dstNamespacedServiceAcc)
 		if destErr != nil {
 			log.Error().Msgf("TrafficTarget %s/%s could not get destination services for service account %s", trafficTargets.Namespace, trafficTargets.Name, dstNamespacedServiceAcc.String())
 			return nil, destErr
 		}
+		destService := destServiceList[0]
```
Comment (shashankram) [inline pkg/catalog/routes.go:245]: would be good to create an issue and reference it wherever these assumptions are being made.

File: pkg/catalog/service.go (hunk 1 above)
Comment (shashankram) [reply pkg/catalog/service.go:15]: Thank you!

File: pkg/catalog/routes.go (hunk 2 above)
Comment (ksubrmnn) [reply pkg/catalog/routes.go:245]: Updated issue #1550

**Pull request merged**